### DISCUSSION TITLE
Rest API: list of values for query parameters

### DISCUSSION THREAD

**Discussion by jeromecambon** [Budibase/budibase#4830] [Help]
Hi,
I have a REST API which require multiple values fro a query parameter.
Someting like:
http://myUrl?q=a&q=b&q=c
This seems to be a common way to give a list of values for a query param.
When I try to do this in Budibase, I can add multiple 'q' parameters in the 'Params' tab, but only the last one is taken into account in the URL.

Is there a way to achieve this?
Thanks!

<img width="1438" alt="Screen Shot 2022-03-07 at 4 10 13 PM" src="https://user-images.githubusercontent.com/17833345/157061041-bd8e366a-a48f-44ae-ab9c-effc3caac21f.png">

**Comment by mjashanks** [comment]
Hey @jeromecambon - I think Budibase's behaviour is correct here - I'm not sure why you would ever want 3 parameters of the same name. 

I'm happy to raise this if you can explain the use of this though. What does the web-server interpret `q` as, in your example above? do you mean that `q` should be an array/list = `[a, b, c]` ?

**Reply by jeromecambon** [reply]
Yes, it is interpreted as a list by the server. As I said, after few search it seems to be a common way of providing a list of values to Rest APIs.

**Reply by mjashanks** [reply]
Thanks! I'll get it logged. In the meantime, I think you should be able to achieve this manually by typing the parameters into the `path`:

![image](https://user-images.githubusercontent.com/3524181/157270572-61accad6-007d-423e-8fd2-2104b4eaef57.png)

**Reply by jeromecambon** [reply]
I already tried this, but unfortunatly only the last instance of the same param is kept. The others are removed as soon as I send the request.

**Reply by mjashanks** [reply]
ah ok, apologies - this is a bug then. Thanks.

**Comment by ghost** [comment]
I'm also experiencing issues with this. The API I'm using has a sort functionality that requires you to send both the field to sort and the direction of the sort as separate query parameters that share the same name. 

For example:
?sort= field:'FieldName'&sort= direction:'desc'
